server/schedule/operator/builder.go

```diff
@@ -484,8 +521,11 @@ func (b *Builder) prepareBuild() (string, error) {
 		}
 	}
 
+	// Although switch witness may have nothing to do with conf change (except switch witness voter to non-witness voter),
+	// the logic here is reused for batch switch.
 	if len(b.toAdd)+len(b.toRemove)+len(b.toPromote) <= 1 && len(b.toDemote) == 0 &&
-		!(len(b.toRemove) == 1 && len(b.targetPeers) == 1) {
+		!(len(b.toRemove) == 1 && len(b.targetPeers) == 1) &&
+		len(b.toWitness)+len(b.toNonWitness)+len(b.toPromoteAfterSwitchToNonWitness) <= 1 {
```

Comment:
Please add more comments to explain this condition.

Reply:
Added, PTAL